S23/b-sai/main.py

```diff
@@ -0,0 +1,43 @@
+
+def to_lowercase(word: str, language: str):
+    """
+    word: str, the string to be converted to lowercase
+    language: str, the language of the string, in BCP 47 format
+    """
+    result = ""
+
+    for idx, letter in enumerate(word):
+
+        lower_letter = letter.lower()
+        if language == 'tr' or language == 'az':
+            if letter == 'I':
+                lower_letter = 'ı'
+        elif language.startswith(('gd', 'gv', 'ga')):
+            if idx == 1 and (letter in ['A', 'E', 'I', 'O', 'U', 'Á', 'É', 'Í', 'Ó', 'Ú', "Ó"] or ord(letter) in [211]) and word[0] in ['n', 't'] and (len(word)-idx >= 2 and ord(word[idx+1]) != 771):
+                lower_letter = "-"+letter.lower()
+        elif language.startswith('el'):
+            if letter == 'Σ' and idx == len(word)-1:
+                lower_letter = 'ς'
+        elif language.startswith(("zh", "th", "ja")):
```

Comment:
It's not going to save a ton of time, but if you aren't making any changes to the letters in these languages -- you don't need to loop through each character. Consider moving this conditional to the top of your loop and exiting early.